commons/src/main/java/org/infinispan/commons/dataconversion/BinaryEncoder.java

```diff
@@ -53,7 +53,7 @@ private static boolean isTypeExcluded(Class<?> type) {
    }
 
    private boolean skipEncoding(Object source) {
-      return isTypeExcluded(source.getClass());
+      return source == null || isTypeExcluded(source.getClass());
```

Comment:
Yeah jboss marshaller has a placeholder short that it would return when it receives a `null`.